modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/utils/AggregatedErrorsCharacteristic.java

```diff
@@ -78,10 +84,46 @@ public List<ErrorModelParser> apply(ComponentAstWithHierarchy componentAstWithHi
     private void handleOperationOtherThanRaiseError(ComponentAst operationAst, List<ErrorModelParser> models,
                                                     List<ComponentAst> hierarchy) {
       Optional<OperationModel> operationModel = operationAst.getModel(OperationModel.class);
-      operationModel.ifPresent(model -> model.getErrorModels().stream().map(MuleSdkErrorModelParser::new)
+      operationModel.ifPresent(model -> model.getErrorModels().stream()
+          .map(errorModel -> createErrorModelParser(errorModel, operationAst))
           .forEach(errorModelParser -> addParserAndMarkIfSuppressed(errorModelParser, models, hierarchy)));
     }
 
+    private MuleSdkErrorModelParser createErrorModelParser(ErrorModel errorModel, ComponentAst operationAst) {
+      final ComponentParameterAst errorMappingsParam = operationAst.getParameter(ERROR_MAPPINGS, ERROR_MAPPINGS_PARAMETER_NAME);
+      if (errorMappingsParam == null) {
+        return new MuleSdkErrorModelParser(errorModel);
+      }
+
+      Optional<List<ErrorMapping>> errorMappingsOpt = errorMappingsParam.<List<ErrorMapping>>getValue().getValue();
+      if (!errorMappingsOpt.isPresent()) {
+        return new MuleSdkErrorModelParser(errorModel);
+      }
+
+      List<ErrorMapping> errorMappings = errorMappingsOpt.get();
+      return applyMappingIsSomeMatches(errorModel, errorMappings);
+    }
+
+    private MuleSdkErrorModelParser applyMappingIsSomeMatches(ErrorModel errorModel, List<ErrorMapping> errorMappings) {
+      return errorMappings.stream()
+          .filter(errorMapping -> doesMappingSourceMatch(errorMapping, errorModel))
+          .map(Aggregator::buildParserFromTarget)
+          .findFirst()
+          .orElseGet(() -> new MuleSdkErrorModelParser(errorModel));
```

Comment:
Have we checked that the `findFirst` here represents how the error mapping logic will actually be applied?
I understand it doesn't really make much sense to have multiple mappings with the same `sourceType`, but it could happen more often with parent relationships, where there is a mapping for a base error type and another mapping for a more specific one.
Should the first one be used? or the more specific one?

Reply:
Yes, it can be seen in this test: https://github.com/mulesoft/mule-ee/blob/master/modules/http-policy/src/test/resources/http-proxy-operation-policy-with-transitive-error-mappings.xml
Where the assertion in the test is for the message "Policy custom error mapping was applied".
I will add a test to cover this case

Reply:
Test added

Reply:
Mmm.. but I wasn't referring to transitive mappings, I meant something like this:

```
<error-mapping sourceType="HEISENBERG:CONNECTIVITY" targetType="MY:MAPPEDCONNECTIVITY"/>
<error-mapping sourceType="HEISENBERG:HEALTH" targetType="MY:MAPPEDHEALTH"/>
```
What would happen if the operation threw `HEISENBERG:HEALTH` which is a child of `HEISENBERG:CONNECTIVITY`?
With the `findFirst` logic, it will be assumed to be mapped to `MY:MAPPEDCONNECTIVITY`, but is that the same thing that will actually happen in runtime? Which makes me wonder if there isn't any logic that we should be reusing.

Reply:
Added a test for this case too